### ISSUE TITLE
net/haproxy: add support for client certificate authentication

### ISSUE THREAD

**Issue by fraenki** [opnsense/plugins#426]
Via [forum post](https://forum.opnsense.org/index.php?topic=6432.0).

Examples:
https://www.haproxy.com/blog/ssl-client-certificate-management-at-application-level/
http://www.loadbalancer.org/blog/client-certificate-authentication-with-haproxy/

**Comment by mohnewald** [comment]
... we are really interested in this feature, too. :-)

**Comment by bluesource** [comment]
here, too!

**Comment by mahescho** [comment]
Count me in, how can I help?

**Comment by fraenki** [comment]
@mahescho Currently nobody is working on this. If you are able to help developing this feature, please submit a pull-request; I'd gladly review your work :)

**Comment by mahescho** [comment]
As I will not use the load balancing feature any way I will have a look at NginX. As I really like HA proxy I will have a look at the source when I find some time. Very basic support should not be to hard to implement.

**Comment by bluesource** [comment]
will there be an approx. timeframe for implementation?
thanks!

**Comment by fraenki** [comment]
@bluesource, in case you missed it:

> Currently nobody is working on this.

No ETA :)

**Comment by mohnewald** [comment]
Would someone be willing/motivated to add this feature for some beer or money? :)

**Comment by fraenki** [comment]
Due to popular demand this feature will be available in the upcoming os-haproxy release 2.10:

![clientcertauth](https://user-images.githubusercontent.com/909706/48034079-26dd1a80-e15e-11e8-86a6-76fa2a4938df.png)


If someone is willing to test it, please apply the current development code:

```
$ opnsense-patch -c plugins 813dfc2b
```

I did a quick test and it seems to work as expected:

```
$ openssl s_client -connect opnsense.example.com:8443 -cert clientcert1.crt -key clientcert1.key
```
